sync/sync.go

```diff
@@ -297,6 +307,138 @@ func (s *Synchronizer) syncBlocks(syncCtx context.Context) {
 	}
 }
 
+func (s *Synchronizer) syncBlocksFeeder(syncCtx context.Context) {
+	streamCtx, streamCancel := context.WithCancel(syncCtx)
+	for {
+		select {
+		case <-streamCtx.Done():
+			streamCancel()
+			return
+		default:
+			s.log.Infow("Fetching from feeder")
+			blocks, stateUpdate, blockCommitments := s.getBlockNumberDetails(uint64(s.latestBlockHeight))
+			newClass, _ := s.fetchUnknownClasses(syncCtx, &stateUpdate)
```

Comment:
You should handle the error

Reply:
I will try to fix this through the manually, if it doesn't work I will switch to the starknetdata interface as you mentioned above!